sds_data_manager/stacks/domain_stack.py

```diff
@@ -0,0 +1,60 @@
+# Installed
+from aws_cdk import Environment, Stack
+from aws_cdk import aws_certificatemanager as acm
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_route53 as route53
+from constructs import Construct
+
+
+class Domain(Stack):
+    """Acquires hosted_zone and certificate"""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        sds_id: str,
+        env: Environment,
+        use_custom_domain: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        sds_id : str
+            Name suffix for stack
+        env : Environment
+        use_custom_domain : bool, Optional
+            Build API Gateway using custom domain
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        self.route_53_role = iam.Role(
+            self,
+            f"Route53Role-{sds_id}",
+            assumed_by=iam.CompositePrincipal(
+                iam.ServicePrincipal("ec2.amazonaws.com")
+            ),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonRoute53FullAccess"
+                )
+            ],
+        )
+
+        if use_custom_domain:
+            self.hosted_zone = route53.HostedZone.from_lookup(
+                self, f"HostedZone-{sds_id}", domain_name="imap-mission.com"
+            )
+
+            self.certificate = acm.Certificate(
+                self,
+                f"Certificate-{sds_id}",
+                domain_name="*.imap-mission.com",
```

Comment:
```suggestion
                domain_name=f"*.{sds_id}.imap-mission.com",
```
If you make the other updates, we might also need to change this cert here, I'm not sure if this is too broad.

Reply:
If we do this change we will need to add more SSL certifications, which is ok but might take more time for me to figure out. Let me know if you want to leave it for now or think it needs to be changed for this pr.

Reply:
I don't know enough about this stuff, my concern would be I deploy a stack with `sds-id="greg"` and create a certificate for `*.imap-mission.com`. What then happens if you want to deploy a stack with `sds-id="laura"`? Are we able to create two different certificates for the same exact domain name? Possibly (because the resource ids would be different), but it'd be worth verifying this too. Did you and @tech3371 deploy your stacks at the same time? That may already give us the answer to this question.

Reply:
I haven't deployed and tested for this PR.